backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/scheduling/policyunits/CpuPinningPolicyUnit.java

```diff
@@ -75,6 +75,10 @@ public CpuPinningPolicyUnit(PolicyUnit policyUnit,
         switch (vm.getCpuPinningPolicy()) {
             case MANUAL:
             case RESIZE_AND_PIN_NUMA:
+                if (StringUtils.isEmpty(cpuPinning)) {
+                    // In resize and pin, we may get here on validation - the cpuPinning is empty in that stage.
+                    return hosts;
+                }
```

Comment:
I'm not sure I understand this section at all - we set the pinning for vms with resize-and-pin according to the *scheduled* host on run-vm, right? so the validation below is for migrate vm? in that case please change the comment in line 79

Reply:
yes, we do. but before we select the host we do: canSchedule() for example. We will get here since the policy is resize but the pinning string isn't set. Then, we will try to parse a null. resulting in NPE.

Reply:
yes yes, I got that part - so would the following explain it correctly:
"The logic below applies to MigrateVm in which we get here with CPU pinning that was set on the source host. On RunVm, we get here before determining the CPU pinning for the VM so the dynamic CPU pinning is not set and hosts should not be filtered by the logic below"
?
if so, please replace the comment in line 79 with something like this that is more informative